src/graph/graph.c

```diff
@@ -1142,14 +1160,74 @@ void Graph_DeleteEdges
 			// there are no additional edges connecting source to destination
 			// remove edge from THE adjacency matrix
 			if(!connected) {
-				M = Graph_GetAdjacencyMatrix(g, false);
-				info = RG_Matrix_removeElement_BOOL(M, src_id, dest_id);
+				Delta_Matrix adj = Graph_GetAdjacencyMatrix(g, false);
+				info = Delta_Matrix_removeElement(adj, src_id, dest_id);
 				ASSERT(info == GrB_SUCCESS);
 			}
+		} else {
+			ME  = Graph_MultiEdgeRelationMatrix(g, r);
+			me_id = CLEAR_MSB(me_id);
+			info = Delta_Matrix_removeElement(ME, me_id, edge_id);
+			ASSERT(info == GrB_SUCCESS);
 		}
 
 		// free and remove edges from datablock.
-		DataBlock_DeleteItem(g->edges, ENTITY_GET_ID(e));
+		DataBlock_DeleteItem(g->edges, edge_id);
+	}
+
+	for (uint i = 0; i < n; i++) {
+		Edge       *e         =  edges + i;
+		int         r         =  Edge_GetRelationID(e);
+		NodeID      src_id    =  Edge_GetSrcNodeID(e);
+		NodeID      dest_id   =  Edge_GetDestNodeID(e);
+		EdgeID      edge_id   =  ENTITY_GET_ID(e);
+
+		M = Graph_GetRelationMatrix(g, r, false);
+
+		GrB_Index me_id;
+		info = Delta_Matrix_extractElement_UINT64(&me_id, M, src_id, dest_id);
+		if(info != GrB_SUCCESS || SINGLE_EDGE(me_id)) continue;
+
+		ME  = Graph_MultiEdgeRelationMatrix(g, r);
+		me_id = CLEAR_MSB(me_id);
+		Delta_MatrixTupleIter it = {0};
+		Delta_MatrixTupleIter_AttachRange(&it, ME, me_id, me_id);
+		GrB_Index last_edge_id;
+		uint count = 0;
+		while (Delta_MatrixTupleIter_next_BOOL(&it, NULL, &last_edge_id, NULL) == GrB_SUCCESS) {
+			count++;
+			if(count == 2) break;
+		}
+
+		if(count == 0) {
+			info = Delta_Matrix_removeElement(M, src_id, dest_id);
+			ASSERT(info == GrB_SUCCESS);
+
+			// see if source is connected to destination with additional edges
+			bool connected = false;
+			int relationCount = Graph_RelationTypeCount(g);
+			for(int j = 0; j < relationCount; j++) {
+				if(j == r) continue;
+				Delta_Matrix r = Graph_GetRelationMatrix(g, j, false);
+				info = Delta_Matrix_extractElement_BOOL(NULL, r, src_id, dest_id);
+				if(info == GrB_SUCCESS) {
+					connected = true;
+					break;
+				}
+			}
+
+			// there are no additional edges connecting source to destination
+			// remove edge from THE adjacency matrix
+			if(!connected) {
+				Delta_Matrix adj = Graph_GetAdjacencyMatrix(g, false);
+				info = Delta_Matrix_removeElement(adj, src_id, dest_id);
+				ASSERT(info == GrB_SUCCESS);
+			}
```

Comment:
Haven't we dealt with this above ?
In my eyes this entire loop deals with the transition from MULTI-EDGE to either a SINGLE-EDGE or NO-EDGE.
Which I think might be better dealt with a new encapsulation of the Relationship matrix.
e.g. Add entry to a Relation-Matrix / Remove an entry from a Relation-Matrix.